lib/os_places_api/client.rb

```diff
@@ -18,10 +18,16 @@ def locations_for_postcode(postcode)
       validate_response_code(response)
 
       begin
-        json = JSON.parse(response)
+        json = JSON.parse(response.body)
         raise UnexpectedResponse if json["results"].nil?
 
-        json["results"]
+        json["results"].map do |result|
+          Location.new(postcode: result["DPA"]["POSTCODE"],
+                       address: result["DPA"]["ADDRESS"],
+                       latitude: result["DPA"]["LAT"],
+                       longitude: result["DPA"]["LNG"],
```

Comment:
I wonder if `dig` would be better here?  E.g. `longitude: result.dig("DPA", "LNG")`?

Reply:
@hannako @brucebolt Thank you both - I have refactored it, please take a look.